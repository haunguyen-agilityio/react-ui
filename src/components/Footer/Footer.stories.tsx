//Libraries
import { ComponentStory, ComponentMeta } from '@storybook/react'

// Components
import Footer from '../Footer'

export default {
  title: 'Components/Footer',
  component: Footer,
} as ComponentMeta<typeof Footer>

const Template: ComponentStory<typeof Footer> = (args) => <Footer {...args} />

export const Primary = Template.bind({})
Primary.args = {
  isDisabled: false,
  isSelected: false,
}

export const Disabled = Template.bind({})
Disabled.args = {
  isDisabled: true,
  isSelected: false,
}

export const Selected = Template.bind({})
Selected.args = {
  isDisabled: false,
  isSelected: true,
}
