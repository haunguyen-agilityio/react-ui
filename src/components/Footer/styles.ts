// Libraries
import { styled } from '@mui/system'
import ListItemButton from '@mui/material/ListItemButton'

// Themes
import theme from '../../themes'

export const StyledListItemButton = styled(ListItemButton)({
  textAlign: 'center',
  color: theme.palette.common.white,
  marginRight: theme.spacing(0.5),
  padding: theme.spacing(1.5, 0),
  '&:hover': {
    backgroundColor: theme.palette.grey[400],
    color: theme.palette.common.black,
  },
  '&.Mui-selected': {
    backgroundColor: theme.palette.grey[400],
    color: theme.palette.common.black,
  },
})

export const StyledFooter = styled('footer')({
  color: theme.palette.common.white,
  backgroundColor: theme.palette.grey[500],
  padding: theme.spacing(0, 4.2),
})
