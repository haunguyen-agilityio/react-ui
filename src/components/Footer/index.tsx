// Libraries
import Container from '@mui/material/Container'
import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Box from '@mui/material/Box'
import Stack from '@mui/material/Stack'
import Typography from '@mui/material/Typography'

// Styles
import { StyledListItemButton, StyledFooter } from './styles'

interface IFooter {
  isDisabled?: boolean
  isSelected?: boolean
}

const Footer = ({ isDisabled = false, isSelected = false }: IFooter) => {
  return (
    <StyledFooter>
      <Container maxWidth={false} disableGutters={true}>
        <List disablePadding={true}>
          <Stack direction='row' alignItems='center'>
            <Box flex={3}>
              <ListItem disablePadding={true}>
                <Typography>&copy; 2020 New York Digital Investment Group LLC</Typography>
              </ListItem>
            </Box>
            <Box flex={1} alignItems='center'>
              <ListItem disablePadding={true}>
                <StyledListItemButton
                  data-testid='supportButton'
                  disabled={isDisabled}
                  selected={isSelected}
                >
                  {/* TODO: Update Link for Support */}
                  <ListItemText primary='Support' />
                </StyledListItemButton>
                <StyledListItemButton
                  data-testid='privacyButton'
                  disabled={isDisabled}
                  selected={isSelected}
                >
                  {/* TODO: Update Link for Privacy Policy */}
                  <ListItemText primary='Privacy Policy' />
                </StyledListItemButton>
                <StyledListItemButton
                  data-testid='termsButton'
                  disabled={isDisabled}
                  selected={isSelected}
                >
                  {/* TODO: Update Link for Terms of Service */}
                  <ListItemText primary='Terms of Service' />
                </StyledListItemButton>
              </ListItem>
            </Box>
          </Stack>
        </List>
      </Container>
    </StyledFooter>
  )
}

export default Footer
