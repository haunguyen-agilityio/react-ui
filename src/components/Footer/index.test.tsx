// Libraries
import { screen, render } from '@testing-library/react'
import * as renderer from 'react-test-renderer'

// Components
import Footer from '../Footer'

describe('Footer', () => {
  const defaultProps = {}
  it('Component [Footer] matches DOM Snapshot', () => {
    const component = renderer.create(<Footer {...defaultProps} />)

    expect(component).toMatchSnapshot()
  })

  it('Component [Input] should render correctly', () => {
    render(<Footer {...defaultProps} />)

    expect(screen.getByTestId('supportButton')).toHaveTextContent('Support')
    expect(screen.getByTestId('privacyButton')).toHaveTextContent('Privacy Policy')
    expect(screen.getByTestId('termsButton')).toHaveTextContent('Terms of Service')
  })
})
