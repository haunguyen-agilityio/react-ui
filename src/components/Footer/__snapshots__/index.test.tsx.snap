// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component [Footer] matches DOM Snapshot 1`] = `
<footer
  className="css-1a04utd"
>
  <div
    className="MuiContainer-root MuiContainer-disableGutters css-idqxm-MuiContainer-root"
  >
    <ul
      className="MuiList-root css-1mk9mw3-MuiList-root"
    >
      <div
        className="css-1d9cypr-MuiStack-root"
      >
        <div
          className="MuiBox-root css-ieri5n"
        >
          <li
            className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            disabled={false}
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Â© 2020 New York Digital Investment Group LLC
            </p>
          </li>
        </div>
        <div
          className="MuiBox-root css-w5t042"
        >
          <li
            className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            disabled={false}
          >
            <div
              className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1bpq8db-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="supportButton"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Support
                </span>
              </div>
            </div>
            <div
              className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1bpq8db-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="privacyButton"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Privacy Policy
                </span>
              </div>
            </div>
            <div
              className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1bpq8db-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="termsButton"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Terms of Service
                </span>
              </div>
            </div>
          </li>
        </div>
      </div>
    </ul>
  </div>
</footer>
`;
